PHPLint/phpoptions.cpp

```diff
@@ -0,0 +1,38 @@
+#include "phpoptions.h"
+#include "globals.h"
+
+PhpOptions::PhpOptions()
+    : clConfigItem("PHPConfigurationData")
+    , m_phpExe("")
+{
+}
+
+PhpOptions::~PhpOptions() {}
+
+void PhpOptions::FromJSON(const JSONElement& json)
+{
+    m_phpExe = json.namedObject("m_phpExe").toString(m_phpExe);
+    if (m_phpExe.IsEmpty()) {
+        wxFileName phpExe;
+        clFindExecutable("php", phpExe);
+        m_phpExe = phpExe.GetFullPath();
+    }
+}
+
+JSONElement PhpOptions::ToJSON() const
+{
+    JSONElement element = JSONElement::createObject(GetName());
+    return element;
+}
+
+PhpOptions& PhpOptions::Load()
+{
+    clConfig config("php.conf");
+    config.ReadItem(this);
+    return *this;
+}
+
+PhpOptions& PhpOptions::Save()
+{
+    return *this;
+}
```

Comment:
The save function is a stub exactly to avoid the chance of it overwriting php.conf.

I tried simply including php_configuration_data.h and adding "${CL_SRC_ROOT}/codelitephp/PHPParser" as include path in the cmake files. This does compile, but at the end the application crashes with the following error:

[22:55:44:466 ERR] Failed to create single instance socket: CreateServer: bind operation failed: Address already in use
codelite: symbol lookup error: /usr/lib/x86_64-linux-gnu/codelite/CodeFormatter.so: undefined symbol: _ZN20PHPConfigurationDataC1Ev

Am I missing something or should it be solved in another way?